types/heimdalljs/index.d.ts

```diff
@@ -0,0 +1,3 @@
+declare module 'heimdalljs' {
+  export default any
+}
```

Comment:
Yes, this is bad and I feel bad. We should fix the types publishing of heimdalljs@0.2.x if we actually want good types here.

I'm not sure I want to shave that particular yak at the moment...

Reply:
I mean, the existence of OneShot also seems bad and I feel bad. 😅

Do you have any sense for whether BROCOLLI_ENABLED_MEMOIZE is going to get widely enabled? It would let us drop OneShot.

We could also profile how much OneShot is really saving us on rebuilds. I think it mattered when I added it, but a lot has changed since then.

Reply:
> Do you have any sense for whether BROCOLLI_ENABLED_MEMOIZE is going to get widely enabled? It would let us drop OneShot.

The main problem is that we need to provide a reasonable escape hatch that isn't "restart the server". @stefanpenner and I have chatted about a few options, I think that the best one was to add a way to trigger a full rebuild (e.g. ignoring the memoization for that specific rebuild). The two ways we thought of doing that were something like `touch tmp/perform-full-rebuild` (just touch some file that we could monitor) or sending a signal to the running process (e.g. `SIGUSR1`) that would force it to full rebuild.

> We could also profile how much OneShot is really saving us on rebuilds. I think it mattered when I added it, but a lot has changed since then.

I suspect it still saves us quite a bit of time, and will until we start moving the needle on usage of `@embroider/addon-shim`.